### PR TITLE
[ISSUE #10662]Prometheus-sd add namespace and service api

### PR DESCRIPTION
**Pull request by Joey777210**
Please do not create a Pull Request without creating an issue first.

#10662

## What is the purpose of the change
Promethesu sd api returns all service in once. But in lot scenes, Nacos use by alot systems. Needs isolation.
Add two interfaces, and specify Namespace and Service.

## Brief changelog

Add two interfaces, and specify Namespace and Service.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [Y] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [Y] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [Y] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [Y] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [Y] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by KomachiSion** [comment]
As I comment in issue, does prometheus sd protocol can call the specified api?

And when you implement, Please pay attention to the auth feature.

File: prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java [hunk 1]
```diff
@@ -73,26 +72,73 @@ public ResponseEntity metric() throws NacosException {
...
+    }
+    
+    
+    /**
+     * Get service instances from designated namespace.
+     *
+     * @throws NacosException NacosException.
+     */
+    @GetMapping(value = ApiConstants.PROMETHEUS_CONTROLLER_NAMESPACE_PATH, produces = "application/json; charset=UTF-8")
+    public ResponseEntity metricNamespace(@PathVariable("namespace") String namespace) throws NacosException {
```
Comment (KomachiSion) [inline prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java:91]: Should we use namespaceId?  nacos has namespaceShowName and namespaceId, directly use namespace may be let users wrong input namespaceShowName.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java (hunk 1 above)
Comment (Joey777210) [reply prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java:91]: I thought "namespace" is "namespaceId"..  Isn't it?   If it is, may be variable name in Service.class should be changed, If not, I can try find a way to get namespaceId....

Comment (KomachiSion) [reply prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java:91]: Yes it's, but I mean users may be confused and may use `showName`

Comment (KomachiSion) [reply prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java:91]: Only change the parameters name

Comment (Salhazry) [reply prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java]: > Should we use namespaceId?  nacos has namespaceShowName and namespaceId, directly use namespace may be let users wrong input namespaceShowName.

**Review by KomachiSion — APPROVED** [review]

**Comment by KomachiSion** [comment]
It seems ci in prometheus module can't pass.

**Comment by KomachiSion** [comment]
> It seems ci in prometheus module can't pass.

```
Error:    PrometheusControllerTest.testMetricNamespace:112 » IllegalArgument Not enough ...
Error:    PrometheusControllerTest.testMetricNamespaceService:125 » IllegalArgument Not ...
``` 
Please fix the CI first.

**Review by KomachiSion — APPROVED** [review]

**Comment by KomachiSion** [comment]
Welcome to submit document to nacos-group

**Pull request merged**